src/scheduler/tasks/BillPendingTransactionTask.ts

```diff
@@ -0,0 +1,152 @@
+                    $match: {
+                      'stop': { $exists: true },
+                      'stop.extraInactivityComputed': false,
+                      'billingData.stop.status': BillingStatus.PENDING
+                    }
+                  },
+                  {
+                    $project: { '_id': 1 }
+                  }
+                ]).toArray();
+              if (!Utils.isEmptyArray(transactionsMDB)) {
+                await Logging.logInfo({
+                  tenantID: tenant.id,
+                  action: ServerAction.BILLING_BILL_PENDING_TRANSACTION,
+                  module: MODULE_NAME, method: 'processTenant',
+                  message: `The billing of ${transactionsMDB.length} transactions is pending`,
+                });
+                for (const transactionMDB of transactionsMDB) {
+                  const transactionLock = await LockingHelper.acquireBillPendingTransactionLock(tenant.id, transactionMDB._id);
+                  if (transactionLock) {
+                    try {
+                    // Get Transaction
+                      const transaction = await TransactionStorage.getTransaction(tenant, transactionMDB._id, { withUser: true });
+                      if (!transaction) {
+                        await Logging.logError({
+                          tenantID: tenant.id,
+                          action: ServerAction.BILLING_BILL_PENDING_TRANSACTION,
+                          module: MODULE_NAME, method: 'processTenant',
+                          message: `Transaction '${transactionMDB._id}' not found`,
+                        });
+                        continue;
+                      }
+                      // Get Charging Station
+                      const chargingStation = await ChargingStationStorage.getChargingStation(tenant, transaction.chargeBoxID);
+                      if (!chargingStation) {
+                        await Logging.logError({
+                          tenantID: tenant.id,
+                          action: ServerAction.BILLING_BILL_PENDING_TRANSACTION,
+                          module: MODULE_NAME, method: 'processTenant',
+                          message: `Charging Station '${transaction.chargeBoxID}' not found`,
+                        });
+                        continue;
+                      }
+                      // Check for the last transaction
+                      const lastTransaction = await TransactionStorage.getLastTransactionFromChargingStation(tenant, transaction.chargeBoxID, transaction.connectorId);
+                      if (transaction.id === lastTransaction?.id) {
+                        // Avoid conflict with a session which is still in progress
+                        const connector = Utils.getConnectorFromID(chargingStation, transaction.connectorId);
+                        if (connector.status !== ChargePointStatus.AVAILABLE) {
```

Comment:
Do not check the connector's status but it's transaction ID:
`      if (connector.currentTransactionID === transaction.id) {`